compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SDLFileGenerator.java

```diff
@@ -0,0 +1,391 @@
+    private static final String ROOT_TYPE_FORMAT = "schema {%n%s%n}";
+    private static final String DIRECTIVE_TYPE_FORMAT = "%sdirective @%s%s on %s";
+    private static final String INTERFACE_TYPE_FORMAT = "%sinterface %s%s %s";
+    private static final String UNION_TYPE_FORMAT = "%sunion %s%s";
+    private static final String SCALAR_TYPE_FORMAT = "%sscalar %s";
+    private static final String OBJECT_TYPE_FORMAT = "%stype %s%s %s";
+    private static final String ENUM_TYPE_FORMAT = "%senum %s %s";
+    private static final String INPUT_TYPE_FORMAT = "input %s %s";
+    private static final String FIELD_FORMAT = "%s  %s%s: %s%s";
+    private static final String FIELD_BLOCK_FORMAT = "{%n%s%n}";
+    private static final String ARGS_FORMAT = "(%s)";
+    private static final String DESC_FORMAT = "%s%n";
+    private static final String DEPRECATE_FORMAT = "%s(reason: \"%s\")";
+    private static final String COMMENT_FORMAT = "%s# %s";
+    private static final String IMPLEMENT_FORMAT = " implements %s";
+    private static final String POSSIBLE_TYPE_FORMAT = " = %s";
+    private static final String INPUT_FIELD_FORMAT = "  %s: %s";
+    private static final String ARGS_TYPE_FORMAT = "%s = %s";
+    private static final String ARGS_VALUE_FORMAT = "%s: %s";
+    private static final String ENUM_VALUE_FORMAT = "  %s%s";
+    private static final String NON_NULL_FORMAT = "%s!";
+    private static final String LIST_FORMAT = "[%s]";
+    private static final String QUERY_FORMAT = "  query: %s";
+    private static final String MUTATION_FORMAT = "  mutation: %s";
+    private static final String SUBSCRIPTION_FORMAT = "  subscription: %s";
+    private static final String DEPRECATE = " @deprecated";
+
+    //Schema delimiters
+    private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+    private static final String INDENTATION = "  ";
+    private static final String EMPTY_STRING = "";
+    private static final String COMMA_SIGN = ", ";
+    private static final String AND_SIGN = " & ";
+    private static final String PIPE_SIGN = "|";
+
+    public SDLFileGenerator(Schema schema, int serviceHashCode, Project project) {
+        this.schema = schema;
+        this.serviceHashCode = serviceHashCode;
+        this.project = project;
+    }
+
+    public void generate() {
+        String schema = getSDLSchemaString();
+        writeFile(schema);
+    }
+
+    private String getSDLSchemaString() {
+        String rootType = createRootType();
+        String directives = getDirectives();
+        String types = getTypes();
+        return getFormattedString(SCHEMA_FORMAT, createDescription(this.schema.getDescription()), rootType, directives,
+                                  types);
+    }
+
+    private String createRootType() {
+        List<String> rootOperations = new ArrayList<>();
+        Type query = this.schema.getQueryType();
+        if (query != null) {
```

Comment:
Is this check necessary? If there's not query type, the schema will not be generated anyway, right?